### PR TITLE
Move additional helpers from the factory

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

~~Builds on https://github.com/kubernetes/kubernetes/pull/63927~~

cc @deads2k @soltysh

### CODE REVIEW

File: pkg/kubectl/polymorphichelpers/helpers.go [hunk 1]
```diff
@@ -233,3 +239,60 @@ func SelectorsForObject(object runtime.Object) (namespace string, selector label
 
 	return namespace, selector, nil
 }
+
+// attachablePodForObject returns the pod to which to attach given an object.
+func attachablePodForObject(restClientGetter genericclioptions.RESTClientGetter, object runtime.Object, timeout time.Duration) (*api.Pod, error) {
```
Comment (deads2k) [inline pkg/kubectl/polymorphichelpers/helpers.go]: make a separate file for these.  makes it obvious which ones are missing tests.  you don't need to add a test, but let's make our debt obvious

**Comment by deads2k** [comment]
file structure change, rebase, squash,

/lgtm

**Comment by juanvallejo** [comment]
@deads2k 
> file structure change, rebase, squash,

Thanks, done

**Pull request merged**